src/lib.rs

```diff
@@ -0,0 +1,275 @@
+use log::trace;
+use std::{collections, fs, path};
+
+pub struct FragmentScanner {
+    dirs: Vec<path::PathBuf>,
+}
+
+impl FragmentScanner {
+    /// Returns a new FragmentScanner, initialized with a vector of directory paths to scan for
+    /// configuration fragments.
+    ///
+    /// # Arguments
+    ///
+    /// * `base_dirs` - Vector holding base components of directories where configuration fragments
+    ///                 are located.
+    /// * `shared_path` - Common relative path from each entry in `base_dirs` to the directory
+    ///                   holding configuration fragments.
+    ///
+    /// `shared_path` is concatenated to each entry in `base_dirs` to form the directory paths to
+    /// scan.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let base_dirs = vec![
+    ///     "/usr/lib".to_string(),
+    ///     "/run".to_string(),
+    ///     "/etc".to_string(),
+    /// ];
+    ///
+    /// use liboverdrop::FragmentScanner;
+    /// let od_cfg = FragmentScanner::new(base_dirs, "my-crate/config.d");
+    /// ```
+    pub fn new(
+        base_dirs: Vec<String>,
+        shared_path: &str,
+    ) -> Self {
+        let mut dirs = Vec::with_capacity(base_dirs.len());
+        for bdir in base_dirs {
+            let mut dpath = path::PathBuf::from(bdir);
+            dpath.push(shared_path.clone());
+            dirs.push(dpath);
+        }
+        Self { dirs }
+    }
+
+    /// Scan unique configuration fragments from the set configuration directories. Returns a
+    /// `collections::BTreeMap` indexed by configuration fragment filename, holding the path where
+    /// the unique configuration fragment is located.
+    ///
+    /// Configuration fragments are stored in the `BTreeMap` in alphanumeric order by filename.
+    /// Configuration fragments existing in directories that are scanned later override fragments
+    /// of the same filename in directories that are scanned earlier.
+    ///
+    /// # Arguments
+    ///
+    /// * `allow_hidden` - Whether to allow scanning hidden files (files prefixed with '.').
+    /// * `allowed_extensions` - If specified, only scan files that have an extension listed in
+    ///                          `allowed_extensions`. If not specified, then any extensions are
+    ///                          allowed.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let base_dirs = vec![
+    ///     "/usr/lib".to_string(),
+    ///     "/run".to_string(),
+    ///     "/etc".to_string(),
+    /// ];
+    /// let allowed_extensions = vec![
+    ///     "toml",
+    /// ];
+    ///
+    /// use liboverdrop::FragmentScanner;
+    /// let od_cfg = FragmentScanner::new(base_dirs, "my-crate/config.d");
+    /// let fragments = od_cfg.scan_unique_files(false, Some(allowed_extensions));
+    /// ```
+    pub fn scan_unique_files(
```

Comment:
`scan` or `scan_unique` is already enough, I think.

Reply:
Agreed, fixed using `scan`. Initially thought it was worth keeping `_unique` in case slightly different scanning functions are later implemented (e.g. if one didn't want to override and wanted to append every configuration file from each directory, a function like that could be added named something like `scan_append()` or `scan_all()`). It makes sense naming the common case using overrides as the short `scan()` function.